crates/sui-network/src/state_sync/mod.rs

```diff
@@ -101,7 +102,7 @@ use self::metrics::Metrics;
 #[derive(Clone, Debug)]
 pub struct Handle {
     sender: mpsc::Sender<StateSyncMessage>,
-    checkpoint_event_sender: broadcast::Sender<VerifiedCheckpoint>,
+    synced_checkpoint_notifier: Arc<Notify>,
```

Comment:
I would greatly prefer it if you would not change this from the existing channel. Changing this would dramatically reduce the debug ability and testability of state-sync

Reply:
If you no longer want to rely on the actual checkpoint sent on the channel to the executor, then i would recommend just ignoring the value in the executor. Or we could also go about adding the notifier in addition to the event sender.

Reply:
Makes sense. Reverted that change.